src/Util/ClassSourceManipulator.php

```diff
@@ -64,14 +65,21 @@ public function __construct(
         private bool $overwrite = false,
         private bool $useAttributesForDoctrineMapping = true,
     ) {
-        $this->lexer = new Lexer\Emulative([
-            'usedAttributes' => [
-                'comments',
-                'startLine', 'endLine',
-                'startTokenPos', 'endTokenPos',
-            ],
-        ]);
-        $this->parser = new Parser\Php7($this->lexer);
+        if (class_exists(PhpVersion::class)) {
```

Comment:
For the parts of the code that are for v4 of the parser, could you add a little:

```
/* @legacy Support for nikic/php-parser v4 */
```

Reply:
Should I also add this comment to the property `Symfony\Bundle\MakerBundle\Util\ClassSourceManipulator::$lexer`? Because it will be not needed anymore after v4 support is dropped.